### PR TITLE
Add source files for coverage run

### PR DESCRIPTION
**Pull request by JamesWTruher**
A number of tests require the sources to be present in order to work correctly.
During cleanup be sure to remove any lingering powershell.exe processes because subsequent runs will not be able to update the test binaries.

### CODE REVIEW

File: test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1 [hunk 1]
```diff
@@ -196,14 +196,60 @@ try
...
+        {
+            remove-item -force -recurse "${outputBaseFolder}/.git"
+        }
+        if ( Test-Path "$outputBaseFolder/src" )
+        {
+            remove-item -force -recurse "${outputBaseFolder}/src"
+        }
+        Write-LogPassThru -Message "initializing repo in $outputBaseFolder"
+        & $gitexe init
+        Write-LogPassThru -Message "git operation return $LASTEXITCODE"
```
Comment (adityapatwardhan) [inline test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1]: Maybe `"git operation 'init' return $LASTEXITCODE"`

File: test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1 [hunk 2]
```diff
@@ -196,14 +196,60 @@ try
...
+
+        Write-LogPassThru -Message "adding remote"
+        & $gitexe remote add origin https://github.com/PowerShell/PowerShell
+        Write-LogPassThru -Message "git operation return $LASTEXITCODE"
```
Comment (adityapatwardhan) [inline test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1]: Maybe `"git operation 'remote add' return $LASTEXITCODE"`

File: test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1 [hunk 3]
```diff
@@ -196,14 +196,60 @@ try
...
+
+        Write-LogPassThru -Message "setting sparse-checkout"
+        & $gitexe config core.sparsecheckout true
+        Write-LogPassThru -Message "git operation return $LASTEXITCODE"
```
Comment (adityapatwardhan) [inline test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1]: similar to above

File: test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1 [hunk 4]
```diff
@@ -196,14 +196,60 @@ try
...
+
+        Write-LogPassThru -Message "pulling sparse repo"
+        "src" | out-file -encoding ascii .git\info\sparse-checkout
+        & $gitexe pull origin master
+        Write-LogPassThru -Message "git operation return $LASTEXITCODE"
```
Comment (adityapatwardhan) [inline test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1]: similar to above

File: test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1 [hunk 5]
```diff
@@ -196,14 +196,60 @@ try
...
+
+        Write-LogPassThru -Message "checkout commit $commitId"
+        & $gitexe checkout $commitId
+        Write-LogPassThru -Message "git operation return $LASTEXITCODE"
```
Comment (adityapatwardhan) [inline test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1]: similar to above

**Review by adityapatwardhan — APPROVED** [review]
LGTM with minor comments

File: test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1 (hunk 5 above)
Comment (JamesWTruher) [reply test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1]: yah - cut and paste. I've updated the log messages

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**